From ce3fb22bda6dd50248f8f0c5abe207de2f48a3b6 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Mon, 15 Jul 2024 12:51:22 +0300
Subject: [PATCH] [SW-192799] gaudi3: sec: remove the sync manager rr config

As per SoC online, Sync manager doesn't have RR. So, removing
RR configuration for sync manager master interface.

Change-Id: Ie7da5ac88d95288f3b5aafeac64bb1af9ded6a5e
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/arch/gaudi3/common/security.c b/arch/gaudi3/common/security.c
index b9c9eb6..14f8e65 100644
--- a/arch/gaudi3/common/security.c
+++ b/arch/gaudi3/common/security.c
@@ -244,25 +244,6 @@
 		.range_en_full_breg = mmHD0_RIF_RTR_CTRL_RR_LBW_BASE +
 			mmMSTR_IF_RR_LBW_PRIV_RANGE_EN_0,
 	},
-	[HL_SEC_ENGINE_SYNC_MNGR] = {
-		.instances_dist = 0,
-		.min_short_breg = mmHD0_SYNC_MNGR_MSTR_IF_RR_LBW_BASE +
-			mmMSTR_IF_RR_LBW_PRIV_RANGE_MIN_SHORT_0,
-		.max_short_breg = mmHD0_SYNC_MNGR_MSTR_IF_RR_LBW_BASE +
-			mmMSTR_IF_RR_LBW_PRIV_RANGE_MAX_SHORT_0,
-		.min_full_breg = mmHD0_SYNC_MNGR_MSTR_IF_RR_LBW_BASE +
-			mmMSTR_IF_RR_LBW_PRIV_RANGE_MIN_0,
-		.max_full_breg = mmHD0_SYNC_MNGR_MSTR_IF_RR_LBW_BASE +
-			mmMSTR_IF_RR_LBW_PRIV_RANGE_MAX_0,
-		.hit_wr_breg = mmHD0_SYNC_MNGR_MSTR_IF_RR_LBW_BASE +
-			mmMSTR_IF_RR_LBW_PRIV_HIT_AW,
-		.hit_rd_breg = mmHD0_SYNC_MNGR_MSTR_IF_RR_LBW_BASE +
-			mmMSTR_IF_RR_LBW_PRIV_HIT_AR,
-		.range_en_short_breg = mmHD0_SYNC_MNGR_MSTR_IF_RR_LBW_BASE +
-			mmMSTR_IF_RR_LBW_PRIV_RANGE_EN_SHORT_0,
-		.range_en_full_breg = mmHD0_SYNC_MNGR_MSTR_IF_RR_LBW_BASE +
-			mmMSTR_IF_RR_LBW_PRIV_RANGE_EN_0,
-	},
 };
 
 static const struct hl_sec_rrs_cfg_hbw hl_sec_hbw_rrs_tbl[] = {
@@ -349,23 +330,6 @@
 		.hit_rd_breg = mmHD0_RIF_RTR_CTRL_RR_HBW_BASE +
 			mmMSTR_IF_RR_HBW_PRIV_HIT_AR,
 		.range_en_full_breg = mmHD0_RIF_RTR_CTRL_RR_HBW_BASE +
-			mmMSTR_IF_RR_HBW_PRIV_RANGE_EN_0,
-	},
-	[HL_SEC_ENGINE_SYNC_MNGR] = {
-		.instances_dist = 0,
-		.min_full.lo = mmHD0_SYNC_MNGR_MSTR_IF_RR_HBW_BASE +
-			mmMSTR_IF_RR_HBW_PRIV_RANGE_MIN_LO_0,
-		.min_full.hi = mmHD0_SYNC_MNGR_MSTR_IF_RR_HBW_BASE +
-			mmMSTR_IF_RR_HBW_PRIV_RANGE_MIN_HI_0,
-		.max_full.lo = mmHD0_SYNC_MNGR_MSTR_IF_RR_HBW_BASE +
-			mmMSTR_IF_RR_HBW_PRIV_RANGE_MAX_LO_0,
-		.max_full.hi = mmHD0_SYNC_MNGR_MSTR_IF_RR_HBW_BASE +
-			mmMSTR_IF_RR_HBW_PRIV_RANGE_MAX_HI_0,
-		.hit_wr_breg = mmHD0_SYNC_MNGR_MSTR_IF_RR_HBW_BASE +
-			mmMSTR_IF_RR_HBW_PRIV_HIT_AW,
-		.hit_rd_breg = mmHD0_SYNC_MNGR_MSTR_IF_RR_HBW_BASE +
-			mmMSTR_IF_RR_HBW_PRIV_HIT_AR,
-		.range_en_full_breg = mmHD0_SYNC_MNGR_MSTR_IF_RR_HBW_BASE +
 			mmMSTR_IF_RR_HBW_PRIV_RANGE_EN_0,
 	},
 };
@@ -745,16 +709,6 @@
 		hl_secure_lbw_rr_gen(rr_idx, min, max, 0, 0, HL_SEC_ENGINE_NW12_DEC, false, enable, 0);
 	}
 
-	if (rr_sel.sync_mngr) {
-		for (die = 0; die <= data->sku; die++) {
-			die1_hdcore_offset = (DIE1_HDCORE_OFFSET * die);
-			for (hdcore = 0; hdcore < GAUDI3_NUM_HDCORES_PER_DIE; hdcore++)
-				hl_secure_lbw_rr_gen(rr_idx, min, max, ((HDCORE_OFFSET * hdcore) +
-						die1_hdcore_offset), 0, HL_SEC_ENGINE_SYNC_MNGR,
-						false, enable, die);
-		}
-	}
-
 	if (true == log_require) {
 		lbw_idx++; /*Increment LBW RR configure logging debug buffer index */
 		lbw_rr_dbg_arr[rr_idx] += 1; /* counting each RR configured how many times */
@@ -838,15 +792,6 @@
 		hl_secure_hbw_rr_gen(rr_idx, min, max, 0, 1, HL_SEC_ENGINE_NRTR, 0);
 	}
 
-	if (rr_sel.sync_mngr) {
-		for (die = 0; die <= data->sku; die++) {
-			die1_hdcore_offset = (DIE1_HDCORE_OFFSET * die);
-			for (hdcore = 0; hdcore < GAUDI3_NUM_HDCORES_PER_DIE; hdcore++)
-				hl_secure_hbw_rr_gen(rr_idx, min, max, ((HDCORE_OFFSET * hdcore) +
-						die1_hdcore_offset), 1, HL_SEC_ENGINE_SYNC_MNGR, die);
-		}
-	}
-
 	if (true == log_require) {
 		hbw_idx++; /*Increment HBW RR configure logging debug buffer index */
 		hbw_rr_dbg_arr[rr_idx] += 1; /* counting each RR configured how many times */
@@ -870,7 +815,6 @@
 		/* Block noc and psoc RR in preboot */
 		rr_sel.psoc = 1;
 		rr_sel.noc = 1;
-		rr_sel.sync_mngr = 1;
 
 		min = RR_HBW_PRIV_RANGE_MIN;
 		max = RR_HBW_PRIV_RANGE_MAX;
@@ -914,7 +858,6 @@
 
 		rr_sel.noc = 1;
 		rr_sel.psoc = 1;
-		rr_sel.sync_mngr = 1;
 
 		/* LBW SPACE: 0x0300007FE0000000 - 0x0300007FFFFFFFFF */
 		min = LBW_BASE;
@@ -926,7 +869,6 @@
 
 	rr_sel.noc = 0;
 	rr_sel.psoc = 0;
-	rr_sel.sync_mngr = 0;
 	rr_sel.pcie = HL_SECURITY_PCIE_MSTR_RR | HL_SECURITY_PCIE_LBW_RR;
 
 	/*
diff --git a/arch/gaudi3/common/security_defs.h b/arch/gaudi3/common/security_defs.h
index 927cd87..52fb4d6 100644
--- a/arch/gaudi3/common/security_defs.h
+++ b/arch/gaudi3/common/security_defs.h
@@ -78,10 +78,8 @@
 	uint8_t psoc;
 	/** HMMU RR selection for security configuration */
 	uint8_t hmmu;
-	/** SYNC_MNGR RR selection for security configuration */
-	uint8_t sync_mngr;
 	/** Reserved for future use */
-	uint8_t reserved[11];
+	uint8_t reserved[12];
 };
 
 #endif  //_PREBOOT_SECURITY_DEF_H_
